lib/nanoc/base/entities/configuration.rb

```diff
@@ -47,6 +51,21 @@ def with_defaults
       self.class.new(new_wrapped)
     end
 
+    def with_environment(env = nil)
+      return self unless @wrapped.key? :environments
+
+      # Set active environment
+      env ||= ENV.fetch 'NANOC_ENVIRONMENT', :default
+
+      # Load given environment configuration
+      env_config = @wrapped[:environments][env.to_sym] || {}
+
+      # Handle specific properties
+      env_config[:output_dir] ||= File.join(@wrapped[:output_dir].to_s, env.to_s)
```

Comment:
I’m not in favour of modifying `output_dir` like this. The output_dir can be specified on a per-environment basis, if necessary, so there’s no need for this logic.